diff --git a/debian/control b/debian/control
index 4f0d70a..6eea48b 100644
--- a/debian/control
+++ b/debian/control
@@ -3,7 +3,7 @@ Section: perl
 Priority: optional
 Maintainer: Proxmox Support Team <support@proxmox.com>
 Build-Depends: asciidoc-dblatex,
-               asciidoc-base (>= 10.2.1-1+deb13u1),
+               asciidoc-base (>= 10.2.1-1),
                debhelper-compat (= 13),
                docbook-xsl,
                git,
