diff --git a/proxmox-api-macro/Cargo.toml b/proxmox-api-macro/Cargo.toml
index ba6f0726..9a6bc9ff 100644
--- a/proxmox-api-macro/Cargo.toml
+++ b/proxmox-api-macro/Cargo.toml
@@ -25,14 +25,6 @@ serde = { workspace = true, features = [ "derive" ] }
 serde_json.workspace = true
 proxmox-section-config.workspace = true
 
-[dev-dependencies.proxmox-schema]
-workspace = true
-features = [ "test-harness", "api-macro" ]
-
-[dev-dependencies.proxmox-router]
-workspace = true
-features = [ "test-harness" ]
-
 # [features]
 # # Used to quickly filter out the serde derive noise when using `cargo expand` for debugging!
 # # Add this in case you need it, but don't commit it (to avoid debcargo picking this up)!
