diff --git a/proxmox-section-config/src/lib.rs b/proxmox-section-config/src/lib.rs
index fec6a423..3e1afc4c 100644
--- a/proxmox-section-config/src/lib.rs
+++ b/proxmox-section-config/src/lib.rs
@@ -433,9 +433,7 @@ impl SectionConfig {
                                         id_property: &Option<String>|
          -> Result<(), Error> {
             for (name, optional, _prop_schema) in schema.properties() {
-                if let Some(id_property) = id_property
-                    && name == id_property
-                {
+                if id_property.as_deref() == Some(name) {
                     // the id_property is the section header, skip for requirement check
                     continue;
                 }
diff --git a/proxmox-section-config/src/typed.rs b/proxmox-section-config/src/typed.rs
index 55cb3c75..5250fd86 100644
--- a/proxmox-section-config/src/typed.rs
+++ b/proxmox-section-config/src/typed.rs
@@ -215,10 +215,10 @@ impl<T> SectionConfigData<T> {
     {
         let removed_value = self.sections.remove(key);
         // only update the order vector if we actually removed something
-        if removed_value.is_some()
-            && let Some(pos) = self.order.iter().position(|k| k.borrow() == key)
-        {
-            self.order.remove(pos);
+        if removed_value.is_some() {
+            if let Some(pos) = self.order.iter().position(|k| k.borrow() == key) {
+                self.order.remove(pos);
+            }
         }
         removed_value
     }
