diff --git a/proxmox-fixed-string/src/lib.rs b/proxmox-fixed-string/src/lib.rs
index 47af54ad..52d38bbb 100644
--- a/proxmox-fixed-string/src/lib.rs
+++ b/proxmox-fixed-string/src/lib.rs
@@ -60,7 +60,7 @@ impl FixedString {
     #[inline]
     pub fn as_str(&self) -> &str {
         // SAFETY: self.buf must be a valid utf-8 string by construction
-        unsafe { str::from_utf8_unchecked(self.as_bytes()) }
+        unsafe { std::str::from_utf8_unchecked(self.as_bytes()) }
     }
 
     /// Returns a reference to the set bytes in the stored buffer
