diff --git a/proxmox-apt-api-types/Cargo.toml b/proxmox-apt-api-types/Cargo.toml
index cd0fe265..de2e5fe7 100644
--- a/proxmox-apt-api-types/Cargo.toml
+++ b/proxmox-apt-api-types/Cargo.toml
@@ -16,7 +16,8 @@ enum-fallback = ["dep:proxmox-fixed-string"]
 [dependencies]
 serde = { workspace = true, features = ["derive"] }
 serde_plain.workspace = true
-proxmox-schema = { workspace = true, features = ["api-macro"] }
+proxmox-schema.workspace = true
+proxmox-api-macro.workspace = true
 proxmox-serde = { workspace = true, features = ["perl"] }
 proxmox-config-digest.workspace = true
 proxmox-fixed-string = { workspace = true, optional = true }
diff --git a/proxmox-apt-api-types/src/lib.rs b/proxmox-apt-api-types/src/lib.rs
index d3f5b3eb..9d3a6868 100644
--- a/proxmox-apt-api-types/src/lib.rs
+++ b/proxmox-apt-api-types/src/lib.rs
@@ -3,7 +3,8 @@ use std::fmt::Display;
 use serde::{Deserialize, Serialize};
 
 use proxmox_config_digest::ConfigDigest;
-use proxmox_schema::{api, const_regex, ApiStringFormat};
+use proxmox_api_macro::api;
+use proxmox_schema::{const_regex, ApiStringFormat};
 
 #[cfg(feature = "enum-fallback")]
 use proxmox_fixed_string::FixedString;
