diff --git a/proxmox-network-types/Cargo.toml b/proxmox-network-types/Cargo.toml
index 1278a10f..307c85dd 100644
--- a/proxmox-network-types/Cargo.toml
+++ b/proxmox-network-types/Cargo.toml
@@ -15,7 +15,7 @@ serde = { workspace = true, features = [ "derive", "std" ] }
 serde_with = "3.8.1"
 thiserror.workspace = true
 
-proxmox-schema = { workspace = true, features = [ "api-macro", "api-types" ], optional = true}
+proxmox-schema = { workspace = true, features = [ "api-types" ], optional = true}
 
 [features]
 default = []
