diff --git a/proxmox-rest-server/Cargo.toml b/proxmox-rest-server/Cargo.toml
index 562ffcf4..23410f6a 100644
--- a/proxmox-rest-server/Cargo.toml
+++ b/proxmox-rest-server/Cargo.toml
@@ -12,7 +12,6 @@ repository.workspace = true
 rust-version.workspace = true
 
 [dev-dependencies]
-proxmox-schema = { workspace = true, features = [ "api-macro" ] }
 tokio = { workspace = true, features = [ "rt-multi-thread", "signal", "process" ] }
 
 [dependencies]
@@ -47,7 +46,7 @@ proxmox-log.workspace = true
 proxmox-network-types.workspace = true
 proxmox-rate-limiter = { workspace = true, optional = true, features = [ "shared-rate-limiter" ] }
 proxmox-router.workspace = true
-proxmox-schema = { workspace = true, features = [ "api-macro", "upid-api-impl" ] }
+proxmox-schema = { workspace = true, features = [ "upid-api-impl" ] }
 proxmox-sys = { workspace = true, features = [ "logrotate", "timer" ] }
 proxmox-time.workspace = true
 proxmox-worker-task.workspace = true
