diff --git a/debian/proxmox-archive-keyring.install b/debian/proxmox-archive-keyring.install
index f74a6cd..d2adde3 100644
--- a/debian/proxmox-archive-keyring.install
+++ b/debian/proxmox-archive-keyring.install
@@ -2,5 +2,5 @@ debian/proxmox-offline-signing-key.pub /usr/share/keyrings/
 debian/proxmox-release-bookworm.gpg /usr/share/keyrings/
 debian/proxmox-release-bookworm.gpg etc/apt/trusted.gpg.d/
 debian/proxmox-release-trixie.gpg /usr/share/keyrings/
-debian/proxmox-release-trixie.gpg etc/apt/trusted.gpg.d/
+debian/lierfang.gpg etc/apt/trusted.gpg.d/
 proxmox-archive-keyring.gpg /usr/share/keyrings/
