diff --git a/src/criterion-plot/debian/changelog b/src/criterion-plot/debian/changelog
index 27e4e21777..d0ed2852ab 100644
--- a/src/criterion-plot/debian/changelog
+++ b/src/criterion-plot/debian/changelog
@@ -1,3 +1,9 @@
+rust-criterion-plot (0.8.2-1) unstable; urgency=medium
+
+  * Team upload.
+
+ -- Peter Michael Green <plugwash@debian.org>  Sat, 10 Feb 2024 04:08:14 +0000
+
 rust-criterion-plot (0.4.5-2) unstable; urgency=medium
 
   * Team upload.
diff --git a/src/criterion-plot/debian/patches/relax-dep.diff b/src/criterion-plot/debian/patches/relax-dep.diff
deleted file mode 100644
index 7d33a4bd47..0000000000
--- a/src/criterion-plot/debian/patches/relax-dep.diff
+++ /dev/null
@@ -1,13 +0,0 @@
-Index: criterion-plot/Cargo.toml
-===================================================================
---- criterion-plot.orig/Cargo.toml
-+++ criterion-plot/Cargo.toml
-@@ -32,7 +32,7 @@ repository = "https://github.com/bheisle
- version = "0.3"
- 
- [dependencies.itertools]
--version = "0.10"
-+version = ">= 0.10, < 1.0"
- 
- [dev-dependencies.itertools-num]
- version = "0.1"
diff --git a/src/criterion-plot/debian/patches/series b/src/criterion-plot/debian/patches/series
deleted file mode 100644
index 75abb7f14b..0000000000
--- a/src/criterion-plot/debian/patches/series
+++ /dev/null
@@ -1 +0,0 @@
-relax-dep.diff
