diff --git a/src/serde-derive/debian/changelog b/src/serde-derive/debian/changelog
index 77498a73ca..0bab6438d4 100644
--- a/src/serde-derive/debian/changelog
+++ b/src/serde-derive/debian/changelog
@@ -1,3 +1,10 @@
+rust-serde-derive (1.0.228-1) unstable; urgency=medium
+
+  * Team upload.
+  * Package serde_derive 1.0.228 from crates.io using debcargo 2.7.6
+
+ -- Proxmox Support Team <support@proxmox.com>  Thu, 19 Mar 2026 00:00:00 +0000
+
 rust-serde-derive (1.0.217-1) unstable; urgency=medium
 
   * Team upload.
