diff --git a/src/criterion/debian/changelog b/src/criterion/debian/changelog
new file mode 100644
index 0000000000..4edb96d55b
--- /dev/null
+++ b/src/criterion/debian/changelog
@@ -0,0 +1,5 @@
+rust-criterion (0.8.2-1) unstable; urgency=medium
+
+  * Package rust-criterion 0.8.2 from crates.io using debcargo
+
+ -- Darren <darrenchang951@gmail.com>  Thu, 19 Mar 2026 00:00:00 +0000
diff --git a/src/criterion/debian/copyright b/src/criterion/debian/copyright
new file mode 100644
index 0000000000..f69bb0263a
--- /dev/null
+++ b/src/criterion/debian/copyright
@@ -0,0 +1 @@
+FIXME fill me in using ./copyright.debcargo.hint as a guide
diff --git a/src/criterion/debian/debcargo.toml b/src/criterion/debian/debcargo.toml
new file mode 100644
index 0000000000..d0ada50e5f
--- /dev/null
+++ b/src/criterion/debian/debcargo.toml
@@ -0,0 +1,4 @@
+overlay = "."
+uploaders = [
+  "Darren <darrenchang951@gmail.com>",
+]
diff --git a/src/criterion/debian/patches/relax-rust-dep.diff b/src/criterion/debian/patches/relax-rust-dep.diff
new file mode 100644
index 0000000000..652f1b1a6c
--- /dev/null
+++ b/src/criterion/debian/patches/relax-rust-dep.diff
@@ -0,0 +1,8 @@
+--- a/Cargo.toml
++++ b/Cargo.toml
+@@ -12,5 +12,4 @@
+ [package]
+ edition = "2021"
+-rust-version = "1.86"
+ name = "criterion"
+ version = "0.8.2"
diff --git a/src/criterion/debian/patches/series b/src/criterion/debian/patches/series
new file mode 100644
index 0000000000..a453f112f9
--- /dev/null
+++ b/src/criterion/debian/patches/series
@@ -0,0 +1 @@
+relax-rust-dep.diff
