--- qrcode.orig/Cargo.toml
+++ qrcode/Cargo.toml
@@ -31,10 +31,6 @@
 name = "qrencode"
 
 [[example]]
-name = "encode_image"
-required-features = ["image"]
-
-[[example]]
 name = "encode_string"
 
 [[example]]
@@ -45,18 +41,9 @@
 name = "encode_pic"
 required-features = ["pic"]
 
-[dependencies.image]
-version = "0.25"
-optional = true
-default-features = false
-
-[dev-dependencies.image]
-version = "0.25"
-
 [features]
 bench = []
 default = [
-    "image",
     "svg",
     "pic",
 ]
