diff --git a/src/PVE/Syscall.pm b/src/PVE/Syscall.pm
index d7b0b3f..dcaa485 100644
--- a/src/PVE/Syscall.pm
+++ b/src/PVE/Syscall.pm
@@ -13,6 +13,7 @@ BEGIN {
 	setns => &SYS_setns,
 	syncfs => &SYS_syncfs,
 	fsync => &SYS_fsync,
+	mknod => &SYS_mknodat, # need mknodat for all platforms!
 	openat => &SYS_openat,
 	close => &SYS_close,
 	mkdirat => &SYS_mkdirat,
diff --git a/src/PVE/Tools.pm b/src/PVE/Tools.pm
index 6e429a8..08313e9 100644
--- a/src/PVE/Tools.pm
+++ b/src/PVE/Tools.pm
@@ -27,6 +27,8 @@ use Text::ParseWords;
 use Time::HiRes qw(usleep gettimeofday tv_interval alarm);
 use URI::Escape;
 use base 'Exporter';
+use File::Basename;
+use Cwd 'realpath';
 
 use PVE::Syscall;
 
@@ -1787,8 +1789,14 @@ sub mkdirat($$$) {
 
 sub mknod($$$) {
     my ($filename, $mode, $dev) = @_;
-	die "https://github.com/jiangcuo/Proxmox-Arm64/issues/58";
-    #return syscall(PVE::Syscall::SYS_mknod, $filename, int($mode), int($dev)) == 0;
+	# Actually, we need mknodat syscall for all platforms!
+	# But we still use mknod name for compatibility.
+	my $dir = dirname($filename);
+	my $dirfd = syscall(&PVE::Syscall::SYS_openat, -100, $dir, 0);
+	die "Can't open dir: $!" if $dirfd < 0;
+    my $result =  syscall(PVE::Syscall::mknod, $dirfd, $filename, int($mode), int($dev)) == 0;
+	syscall(&PVE::Syscall::SYS_close, $dirfd);
+	return $result;
 }
 
 sub fchownat($$$$$) {
