diff --git a/src/main.rs b/src/main.rs
index 799cde8..3bcbbaf 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -2,6 +2,7 @@ use std::{
     ffi::{CStr, CString, OsString},
     fs,
     io::ErrorKind,
+    os::raw::c_char,
     os::unix::{ffi::OsStrExt, fs::PermissionsExt},
     path::{Path, PathBuf},
     sync::Arc,
@@ -370,7 +371,7 @@ fn do_rrd_migration(
         bail!("refusing to migrate metrics for {resource:?} - target already exists and 'force' not set!");
     }
 
-    let mut source: [*const i8; 2] = [std::ptr::null(); 2];
+    let mut source: [*const c_char; 2] = [std::ptr::null(); 2];
     source[0] = file.0.as_ptr();
 
     let target_path = CString::new(target_path.to_str().unwrap()).unwrap();
